### PR TITLE
Make sure `dg/bypass-finals` works with Infection

### PR DESCRIPTION
**Pull request by maks-rafalko**
This PR:

- [x] Allows installing `dg/bypass-finals:^1.4.1` with Infection
- [x] Adds e2e test to check `dg/bypass-finals`  works as expected

Closes https://github.com/infection/infection/issues/1275
Related to https://github.com/dg/bypass-finals/issues/9

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -68,7 +68,7 @@
     },
     "conflict": {
         "phpunit/php-code-coverage": ">9 <9.1.4",
-        "dg/bypass-finals": "*"
+        "dg/bypass-finals": "<0.14.1"
```
Comment (sanmai) [inline composer.json:71]: 1.4.1?

Comment (sanmai) [inline composer.json:71]:
```suggestion
        "dg/bypass-finals": "<1.4.1"
```

File: tests/e2e/Stream_Wrapper_Execution/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Ensures Infection works with `dg/bypass-finals`
+
+* https://github.com/dg/bypass-finals/issues/9
+* https://github.com/infection/infection/issues/1275
+
+## Summary
+
+`dg/bypass-finals` before version 0.14.1 overridden Infection's Stream Wrapper and Infection did not create any Mutants.
+
+This tests ensures that starting from 0.14.1 `dg/bypass-finals` works good and do not "disables" Infection's Stream Wrapper.
```
Comment (sanmai) [inline tests/e2e/Stream_Wrapper_Execution/README.md:10]:
```suggestion
`dg/bypass-finals` before version 1.4.1 overridden Infection's Stream Wrapper and Infection did not create any Mutants.

This tests ensures that starting from 1.4.1 `dg/bypass-finals` works good and do not "disables" Infection's Stream Wrapper.
```

**Review by sanmai — APPROVED** [review]
LGTM apart from few nits.

File: composer.json (hunk 1 above)
Comment (maks-rafalko) [reply composer.json]: oh, nice catch. Thank you

**Comment by maks-rafalko** [comment]
Thank you @sanmai

**Comment by TomPavelec** [comment]
Hello I still can't install both packages `infection/infection dev-master conflicts with dg/bypass-finals <1.4.1`.
I have installed latest dg/bypass-finals, version 1.4.0 (issue was solved in this version). Version 1.4.1 doesn't exist.
So i think conflict version here should be annotated as `"dg/bypass-finals": "<1.4.0"`

**Pull request merged**